posydon/grids/termination_flags.py

```diff
@@ -231,8 +232,52 @@ def infer_interpolation_class(tf1, tf2):
     if tf2 in TF2_POOL_NO_RLO:
         return "no_MT"
     if tf1 in TF1_POOL_STABLE:
+        if (binary_history is not None and history1 is not None):
+            rl_overflow1 = binary_history['rl_relative_overflow_1']
+            rl_overflow2 = binary_history['rl_relative_overflow_2']
+            center_h1 = history1['center_h1']
+            forward = np.logical_and(rl_overflow1>RL_RELATIVE_OVERFLOW_THRESHOLD,
+                                     rl_overflow2<RL_RELATIVE_OVERFLOW_THRESHOLD)
+            reverse = np.logical_and(rl_overflow1<RL_RELATIVE_OVERFLOW_THRESHOLD,
+                                     rl_overflow2>RL_RELATIVE_OVERFLOW_THRESHOLD)
+            reverse_postMS = np.logical_and(reverse,
+                                           center_h1<THRESHOLD_CENTRAL_ABUNDANCE)
+            if (np.max(forward) == True and np.max(reverse_postMS) == True):
+                return "stable_reverse_MT"
+            else:
+                return "stable_MT"
         return "stable_MT"
     if tf1 in TF1_POOL_UNSTABLE:
+        if (binary_history is not None and history1 is not None):
```

Comment:
all below: could you please use different variable names, to indicate that they are only from the last entry in the history and not a list/numpy-array like in the stable RLO part.